test/python/test_quantumprogram.py

```diff
@@ -1532,6 +1532,31 @@ def test_offline(self):
         # SDK will throw ConnectionError on every call that implies a connection
         self.assertRaises(ConnectionError, qp.set_api, FAKE_TOKEN, FAKE_URL)
 
+    def test_reconfig(self):
+        """Test reconfiguring the qobj from 1024 shots to 2048 using
+        reconfig instead of recompile
+        """
+        QP_program = QuantumProgram(specs=self.QPS_SPECS)
+        qr = QP_program.get_quantum_register("qname")
+        cr = QP_program.get_classical_register("cname")
+        qc2 = QP_program.create_circuit("qc2", [qr], [cr])
+        qc2.measure(qr[0], cr[0])
+        qc2.measure(qr[1], cr[1])
+        qc2.measure(qr[2], cr[2])
+        circuits = ['qc2']
+        shots = 1024  # the number of shots in the experiment.
+        backend = 'local_qasm_simulator'
+        qobj = QP_program.compile(circuits, backend=backend, shots=shots)
+        out = QP_program.run(qobj)
+        results = out.get_counts('qc2')
+
+        #change the number of shots
+        qobj = QP_program.reconfig(qobj, shots=2048)
+        out2 = QP_program.run(qobj)
```

Comment:
If the purpose of the test is checking .reconfig() is working properly, the test should check every single parameter, not only `shots`. 
```python
qobj = QP_program.reconfig( qobj, shots=2048, backend='another_backend' , .... )
self.assertEqual( qobj['config']['shots'] , 2048)
self.assertEqual( qobj['config']['backend'] , 'another_backend')
self.assertEqual( qobj .... )
...
```
And there's no need to .run() and test the results (there are other tests which already do that).

Reply:
I thought it would be good to run just in case the reconfig someone breaks the qobj. I also added tests for all the other parameters as suggested.